src/lib/salad/rtree.c

```diff
@@ -916,9 +984,26 @@ rtree_iterator_next(struct rtree_iterator *itr)
 			if (neighbor == NULL)
 				return NULL;
 			rtnt_remove(&itr->neigh_tree, neighbor);
+
+			/* Pagination optimisaion */
```

Comment:
Pagination is about starting an iterator in a particular place. So let's implement it completely in `rtree_search`. Also, we wouldn't need to save anything to the iterator with this approach - just pass the pagination options right to `rtree_search`.